app/models/user.rb

```diff
@@ -8,4 +8,12 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :trackable, :validatable, :confirmable
 
   enum role: [:standard, :premium, :admin]
+
+  def downgrade_user!
+    self.role = 0
+    self.wikis.each do |wiki|
```

Comment:
you could make this more efficient doing `self.wikis.upate_all(private: false)` - the way you have it will make a DB query for each wiki and DB queries are expensive